services/spar/src/Spar/API.hs

```diff
@@ -251,17 +251,18 @@ idpDelete zusr idpid = withDebugLog "idpDelete" (const Nothing) $ do
 
 -- | This handler only does the json parsing, and leaves all authorization checks and
 -- application logic to 'idpCreateXML'.
-idpCreate :: Maybe UserId -> IdPMetadataInfo -> Spar IdP
-idpCreate zusr (IdPMetadataValue raw xml) = idpCreateXML zusr raw xml
+idpCreate :: Maybe UserId -> IdPMetadataInfo -> Maybe SAML.IdPId -> Spar IdP
+idpCreate zusr (IdPMetadataValue raw xml) midpid = idpCreateXML zusr raw xml midpid
 
 -- | We generate a new UUID for each IdP used as IdPConfig's path, thereby ensuring uniqueness.
-idpCreateXML :: Maybe UserId -> Text -> SAML.IdPMetadata -> Spar IdP
-idpCreateXML zusr raw idpmeta = withDebugLog "idpCreate" (Just . show . (^. SAML.idpId)) $ do
+idpCreateXML :: Maybe UserId -> Text -> SAML.IdPMetadata -> Maybe SAML.IdPId -> Spar IdP
+idpCreateXML zusr raw idpmeta mReplaced = withDebugLog "idpCreate" (Just . show . (^. SAML.idpId)) $ do
```

Comment:
`replaced` is confusing to me.   `replaces` sounds better?   when we create this Idp it _replaces_ the one in the query parameter no?

Reply:
or rename the query parameter. Either way works I guess.    or `toReplace` ?

Reply:
3dddc3ebb1395dc43d3d7188390a03d7e3a3a48c

I don't know what that name should be, but at least now I hope it's consistent.

I also found this: 8c0c70b6baa38c1f0f3185b21d34b25b0c0617e6